Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1029,29 +1031,39 @@ The decoded 16-character input string totals 10 bytes, so you know that the vali
+The loop is more clearly visible in the graph view below.
+
+![XOR instruction inside a loop](Images/Chapters/0x05c/loop_1784.png)
+
+A note on XOR encryption, XOR is a very commonly used technique to encrypt information where obfuscation is the goal rather than security. XOR should not be used for any serious encryption, as it can be cracked using frequency analysis. Presence of XOR in such validation logic always require special attention and analysis.
+
+Moving forward, at offset `0x000017dc`, the XOR decoded value obtained from above is being compared against returned value from a sub-function call, at `0x000017e8`.
 
 ![Decoded values being compared](Images/Chapters/0x05c/values_compare_17dc.png)
 
-Even though this doesn't look highly sophisticated, you'd still need to analyze more to completely reverse this check and generate a license key that passes it. Now comes the twist: dynamic symbolic execution enables you to construct a valid key automatically! The symbolic execution engine maps a path between the first instruction of the license check (0x00001760) and the code that prints the "Product activation passed" message (0x00001840) to determine the constraints on each byte of the input string.
+Clearly this function is not complex, and can be analyzed manually, but still remains a cumbersome task. Specially while working on a big code base, time can be a major constraint, and it is desirable to automate such analysis. Dynamic symbolic execution is helpful in exactly such situations. The symbolic execution engine maps a path between the first instruction of the license check (at `0x00001760`) and the code that prints the "Product activation passed" message (at `0x00001840`) to determine the constraints on each byte of the input string.
 
 ![If else Graph](Images/Chapters/0x05c/graph_ifelse_1760.png)
 
-The solver engine then finds an input that satisfies those constraints: the valid license key.
+The constraints obtained from the above steps is passed to a solver engine, which solves the constraints to find an input which satisfies them - a valid license key.
+
+To initialize Angr's symbolic execution engine, few setup steps need to be performed:
 
-You need to provide several inputs to the symbolic execution engine:
+- Load the binary into a `Project`. `Project` is the starting point for any kind of analyses in Angr.
 
-- An address from which execution will start. Initialize the state with the first instruction of the serial validation function. This makes the problem significantly easier to solve because you avoid symbolically executing the Base32 implementation.
+- Pass the address from which analysis should start. In current case, we will initialize the state with the first instruction of the serial validation function. This makes the problem significantly easier to solve because you avoid symbolically executing the Base32 implementation.
```

Comment:
```suggestion
- Pass the address from which the analysis should start. In this case, we will initialize the state with the first instruction of the serial validation function. This makes the problem significantly easier to solve because you avoid symbolically executing the Base32 implementation.
```